printing/proof_diffs.ml

```diff
@@ -219,29 +219,31 @@ let to_tuple : Constr.compacted_declaration -> (Names.Id.t list * 'pc option * '
 
 (* XXX: Very unfortunately we cannot use the Proofview interface as
    Proof is still using the "legacy" one. *)
-let process_goal_concl sigma g : Constr.t * Environ.env =
+let process_goal_concl sigma g : EConstr.t * Environ.env =
   let env  = Goal.V82.env   sigma g in
   let ty   = Goal.V82.concl sigma g in
-  let ty   = EConstr.to_constr sigma ty in
   (ty, env)
 
-let process_goal sigma g : Constr.t reified_goal =
+let process_goal sigma g : EConstr.t reified_goal =
   let env  = Goal.V82.env   sigma g in
   let hyps = Goal.V82.hyps  sigma g in
   let ty   = Goal.V82.concl sigma g in
   let name = Goal.uid g             in
   (* There is a Constr/Econstr mess here... *)
-  let ty   = EConstr.to_constr sigma ty in
   (* compaction is usually desired [eg for better display] *)
+  (* next line copied from Printer.pr_context_limit_compact *)
+  let ctxt      = Termops.compact_named_context (Environ.named_context env) in
+  (* how do I modify to_tuple to get the right type? *)
```

Comment:
@ejgallego I have the conclusion working correctly with this code, but I'm not sure how to modify to_tuple to make it work with `ctxt` above.  Any suggestions?